librad/src/paths.rs

```diff
@@ -72,7 +63,7 @@ impl Paths {
         &self.git_includes_dir
     }
 
-    pub fn all_dirs(&self) -> HashMap<&str, &Path> {
+    fn all_dirs(&self) -> impl Iterator<Item = &Path> {
```

Comment:
What's the reason for this change?

Reply:
I’ve removed `pub` because this method is unused by external consumers and keeping the public API small makes it easier to grasp and harder to inadvertently break.

I’ve replaced `HashMap` with the iterator because for consumers it provides all it needs and it simplifies the implementation.

I’m ok with reverting all those secondary changes. They are not related to the functionality but seemed like nice improvements.

Reply:
Ya, it seems like its uses were just getting the `values` of the `HashMap` anyway, and I believe we usually use the direct accessors in upstream.